recipe/meta.yaml

```diff
@@ -10,37 +10,56 @@ source:
   sha256: 0e928e9a07566b7b8d781169f5fa1bcd9e7ae196317d4a433b692802bed499e6
 
 build:
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
-  number: 0
+  number: 1
+  missing_dso_whitelist:  
+    - /lib64/libpthread.so.0
+    - /lib64/libc.so.6
+    - /lib64/libm.so.6
+    - /lib64/ld-linux-x86-64.so.2
 
-requirements:
-  host:
-    - python >=3.7
-    - hatchling >=1.3.1
-    - jupyterlab >=4.0,<5
-    - pip
-    - hatch-jupyter-builder
-  run:
-    - python >=3.7
-    - jupyter_server >=1.6,<3
-    - traitlets >=5.0,<6
-    - nbconvert >=7.0,<8
-    - pydantic >=1.10,<2
-    - sqlalchemy >=1.0,<2
-    - croniter >=1.4,<2
-    - pytz ==2023.3
-    - fsspec ==2023.6.0
-    - s3fs ==2023.6.0
-    - psutil >=5.9,<6
+outputs:
+  - name: {{ name }}
+    build:
+      noarch: python
+      script: python -m pip install . -vv
+    requirements:
+      host:
+        - python >=3.7,<3.12
+        - hatchling >=1.3.1
+        - jupyterlab >=4.0,<5
+        - pip
+        - hatch-jupyter-builder
+      run:
+        - python >=3.7,<3.12
+        - jupyter_server >=1.6,<3
+        - traitlets >=5.0,<6
+        - nbconvert >=7.0,<8
+        - pydantic >=1.10,<2
+        - sqlalchemy >=1.0,<2
+        - croniter >=1.4,<2
+        - pytz ==2023.3
+        - fsspec ==2023.6.0
+        - s3fs ==2023.6.0
+        - psutil >=5.9,<6
+        - libgcc-ng
+        - libstdcxx-ng
```

Comment:
I see from your PR description that without these lines, the Linux build was failing, but I am skeptical of that because the upstream package on PyPI is identical, so nothing should have changed. Could you investigate this issue further? See either 1) if you can remove these requirements, or 2) if you can elaborate on why these requirements are necessary.

Reply:
Here's an example of failing ci:
- [failing CI run](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=801376&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=21950)
- commit: https://github.com/conda-forge/jupyter_scheduler-feedstock/pull/5/commits/c0ec4fc91262772a3564ad7964f37f025137

Using `outputs` section causes stricter (overlinking) checks. Error is due to: 
- `sqlalchemy` depends on `greenlet`
- `greenlet` has 2 objects ( `/greenlet/tests/_test_extension_cpp.cpython-311-x86_64-linux-gnu.so` and `_test_extension_cpp.cpython-311-x86_64-linux-gnu.so`) that are linked against `libgcc-ng` and `libstdcxx-ng`
- at the same time `libgcc-ng` and `libstdcxx-ng` are not specified as `run` dependencies 

There is no reason for us to ship standard GCC libraries that we don't have direct dependencies on. Adding them to `missing_dso_whitelist` would tell conda build to ignore their linkage information during checking ([docs section](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#whitelisting-shared-libraries)).